src/server/pkg/deploy/cmds/cmds.go

```diff
@@ -162,11 +162,13 @@ func DeployCmd(noMetrics *bool) *cobra.Command {
 			if _, err := base64.StdEncoding.DecodeString(args[2]); err != nil {
 				return fmt.Errorf("storage-account-key needs to be base64 encoded; instead got '%v'", args[2])
 			}
-			tempURI, err := url.ParseRequestURI(opts.EtcdVolume)
-			if err != nil {
-				return fmt.Errorf("Volume URI needs to be a well-formed URI; instead got '%v'", opts.EtcdVolume)
+			if opts.EtcdVolume != "" {
```

Comment:
Do we check that exactly one of `--dynamic-etcd-nodes` and `--static-etcd-volume` are set elsewhere? Just want to make sure that this code doesn't run if both are set (and that we emit an error instead).